public/create_user.php

```diff
@@ -1,21 +1,40 @@
 <?php require_once(dirname(__FILE__)."/../modules/ensure_logged_in.php"); ?>
-
+<?php $user_role = get_current_role()?>
+<?php include "templates/header.php"; ?>
+<?php if($user_role == "admin") { ?>
 <?php
 
 require_once "../modules/models/user.php";
 
 if (isset($_POST['submit'])) {
-    $user = new User();
-    $user->first_name = $_POST['first_name'];
-    $user->last_name = $_POST['last_name'];
-    $user->email = $_POST['email'];
-    $user->student_id = $_POST['student_id'];
-    $user->is_admin = 0;
-    $user->is_instructor = 0;
-    $user->is_ta = 0;
-    $user->is_password_changed = 0;
-    $user->password_digest = password_hash($_POST['password'], PASSWORD_DEFAULT);
+    $create_success = false;
+    if($_POST['role'] == "student") {
+        $user = new User();
+        $user->first_name = $_POST['first_name'];
+        $user->last_name = $_POST['last_name'];
+        $user->email = $_POST['email'];
+        $user->set_role($_POST['role']);
+        $user->student_id = $_POST['student_id'];
+        $user->is_password_changed = 0;
+        $user->password_digest = password_hash($_POST['password'], PASSWORD_DEFAULT);
+    
+    } elseif($_POST['role'] == "ta") {
+        $user = new User();
+        $user->first_name = $_POST['first_name'];
+        $user->last_name = $_POST['last_name'];
+        $user->email = $_POST['email'];
+        $user->set_role("ta");
```

Comment:
shouldnt you also set is_password_changed for ta and instructors as well?

Reply:
oh yeah true